src/URI.js

```diff
@@ -1223,6 +1227,26 @@
   };
 
   // compound accessors
+  p.origin = function(v, build) {
+    var parts;
+
+    if (this._parts.urn) {
+      return v === undefined ? '' : this;
```

Comment:
I have no idea on this one. I'll do whatever you think is best, I basically cloned this from the a nearby function.